src/components/loader/loader.jsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import './loader.scss';
+
+export const LOADING_STATES = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  ERROR: 'error'
+};
+
+const Loader = (props)=>{
+  let data;
+    
+  switch (props.status) {
```

Comment:
yeah, this is great!  I think locating this logic inside a single component makes a lot of sense.